dpc/src/record/tests.rs

```diff
@@ -46,7 +46,7 @@ fn test_record_serialization() {
 
             let sn_nonce_input: [u8; 32] = rng.gen();
             let value = rng.gen();
-            let payload: [u8; 32] = rng.gen();
+            let payload = (0..128).map(|_| rng.gen()).collect::<Vec<u8>>();
```

Comment:
nit: sampling of many values can be done in a more `clippy`-  and potentially performance-friendly way using
```
let mut payload = [0u8; PAYLOAD_SIZE]; // this also avoids the Vec allocation
rng.fill(&mut payload);
```
or
```
(&mut rng).sample_iter(Standard).take(PAYLOAD_SIZE).collect::<Vec<u8>>()
```
note: `Rng::gen` is using the standard distribution internally, while `Rng::fill` uses a uniform one